### PR TITLE
Handle empty hook file or other OSErrors

### LINKED ISSUE #632: Empty hook file causes cryptic error
If you have a pre_gen_project.sh or a post_gen_project.sh file with no data in it, cookiecutter fails with an unhelpful traceback.

```
Traceback (most recent call last):
  File "/usr/local/bin/cookiecutter", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/cookiecutter/cli.py", line 100, in main
    config_file=user_config
  File "/usr/local/lib/python2.7/site-packages/cookiecutter/main.py", line 140, in cookiecutter
    output_dir=output_dir
  File "/usr/local/lib/python2.7/site-packages/cookiecutter/generate.py", line 273, in generate_files
    _run_hook_from_repo_dir(repo_dir, 'pre_gen_project', project_dir, context)
  File "/usr/local/lib/python2.7/site-packages/cookiecutter/generate.py", line 232, in _run_hook_from_repo_dir
    run_hook(hook_name, project_dir, context)
  File "/usr/local/lib/python2.7/site-packages/cookiecutter/hooks.py", line 116, in run_hook
    run_script_with_context(script, project_dir, context)
  File "/usr/local/lib/python2.7/site-packages/cookiecutter/hooks.py", line 101, in run_script_with_context
    run_script(temp.name, cwd)
  File "/usr/local/lib/python2.7/site-packages/cookiecutter/hooks.py", line 73, in run_script
    cwd=cwd
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 656, in __init__
    _cleanup()
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 8] Exec format error
```

### PR DESCRIPTION
**Pull request by jcarbaugh**
Took a guess at the right way to test based on the existing tests. Let me know if anything needs to be changed!

Fixes #632.

### CODE REVIEW

**Comment by jcarbaugh** [comment]
Argh... this error might be a *nix specific thing as the same error does not appear to be thrown on Windows. The tests are failing with `DID NOT RAISE` because Windows will always run a script with the shell, not relying on a shebang to exist. I'm going to skip that test on Windows.

**Comment by audreyr** [comment]
@jcarbaugh I think it looks good, but I'd like a second pair of eyes on this before merging.

Can someone who has subprocess experience give this a second review? Just comment on the diff with anything that needs improvement.

File: tests/test_generate_hooks.py [hunk 1]
```diff
@@ -74,6 +76,46 @@ def test_run_python_hooks_cwd():
     assert os.path.exists('inputpyhooks/python_post.txt')
 
 
+@pytest.mark.usefixtures('clean_system', 'remove_additional_folders')
+def test_empty_hooks():
+
+    # OSError.errno=8 is not thrown on Windows when the script is emtpy
```
Comment (michaeljoseph) [inline tests/test_generate_hooks.py]: **empty**

File: cookiecutter/hooks.py [hunk 2]
```diff
@@ -65,15 +66,23 @@ def run_script(script_path, cwd='.'):
...
+            raise FailedHookException(
+                "Hook script failed (exit status: %d)" % exit_status)
+    except OSError as oe:
+        if oe.errno == errno.ENOEXEC:
+            raise FailedHookException(
+                "Hook script failed, might be an "
+                "empty file or missing a shebang")
         raise FailedHookException(
-            "Hook script failed (exit status: %d)" % exit_status)
+            "Hook script failed (error: %s)" % oe)
```
Comment (michaeljoseph) [inline cookiecutter/hooks.py]: Could you please change these string interpolations to `string.format` style?

**Comment by michaeljoseph** [comment]
Looks fine to me, I picked a few nits though @jcarbaugh.

**Comment by michaeljoseph** [comment]
@jcarbaugh 🔔

**Pull request merged**